nuitka/plugins/standard/NumpyPlugin.py

```diff
@@ -173,47 +196,72 @@ class NumpyPlugin(UserPluginBase):
         UserPluginBase: plugin template class we are inheriting.
     """
 
-    plugin_name = "numpy-plugin"  # Nuitka knows us by this name
+    plugin_name = "numpy"  # Nuitka knows us by this name
 
     def __init__(self):
         self.files_copied = False  # ensure one-time action
+        self.scipy = self.getPluginOptionBool("scipy", False)
 
-    def considerExtraDlls(self, dist_dir, module):
-        """ Copy extra shared libraries for this numpy installation.
+    def onStandaloneDistributionFinished(self, dist_dir):
```

Comment:
This is too late, at the time this is called, dependency analysis has already been done, and e.g. rpaths, etc. may still be wrong. On MacOS this seems very common thing. Also other plugins `onStandaloneDistributionFinished` would rather want to see the already completed thing I guess and not help in finishing it.

Basically I think, we need a pattern, where there is only one module that triggers the extra DLLs, and otherwise nothing is done. Like I changed to Tk plugin:

```
        if not _isTkInterModule(module):
            return ()

        if not isWin32Windows():  # if not Windows notify wrong usage once
            info("tkinter plugin supported on Windows only")
            return ()

```

Maybe we allow plugins to overload one method, e.g. `isResponsibleFor(module)` and then have default implementations that make sure to forward the calls only if that returns `True` and allows to implement this easily.

I believe you switching to that method is mainly driven by the fact that you do not want to be called per module, but only once. But I also kind of like the idea to enable a plugin, and maybe have it complain in the end that it was never relevant, but to not do anything unless you have `import numpy`.